src/index.ts

```diff
@@ -217,6 +217,15 @@ export class Connection {
     this.session = session
 
     const fields = result?.fields ?? []
+    // ensure each field has a type assigned,
+    // the only case it would be omitted is in the case of
+    // NULL due to the protojson spec. NULL in our enum
+    // is 0, and empty fields are omitted from the JSON response,
+    // so we should backfill an expected type.
+    fields.forEach((f) => {
+      f.type = f.type || 'NULL'
+    })
```

Comment:
The `for…of` loop avoids a function closure and is a little faster.
```suggestion
    for (const field of fields) {
      field.type ||= 'NULL'
    }
```

Reply:
TIL both `for...of` and `||=`.